lib/ASTGen/Sources/ASTGen/Exprs.swift

```diff
@@ -244,12 +244,7 @@ extension ASTGenVisitor {
     case .simpleInput(let node):
       result.params = self.generate(closureShorthandParameterList: node)
     case nil:
-      result.params = .createParsed(
-        self.ctx,
-        leftParenLoc: nil,
-        parameters: BridgedArrayRef(),
-        rightParenLoc: nil
-      )
+      result.params = nil
```

Comment:
Creating empty parameter list was a mistake. `generate(closureExpr:)` will synthesize anonymous closure parameter according to the `$n` usages.